class-tgm-plugin-activation.php

```diff
@@ -1128,6 +1101,50 @@ public function _get_plugin_data_from_name( $name, $data = 'slug' ) {
 		}
 
 		/**
+		 * Retrieve a link to a plugin information page
+		 *
+		 * @since 2.5.0
+		 *
+		 * @param string $slug
+		 *
+		 * @return string
+		 */
+		public function get_plugin_info_link( $slug ) {
+			$link = '';
+
+			if ( ! empty( $this->plugins[ $slug ]['external_url'] ) && preg_match( self::EXT_REPO_REGEX, $this->plugins[ $slug ]['external_url'] ) ) {
+				$link = sprintf(
+					'<a href="%1$s" target="_blank">%2$s</a>',
+					esc_url( $this->plugins[ $slug ]['external_url'] ),
+					esc_html( $this->plugins[ $slug ]['name'] )
```

Comment:
True, same for abstracting the whole notices functionality to it's own class. There's plenty to do, but not everything needs to be done now.